postprocessing/janus-pp-rec.c

```diff
@@ -476,19 +499,30 @@ int main(int argc, char *argv[])
 			continue;
 		}
 		/* Only read RTP header */
-		bytes = fread(prebuffer, sizeof(char), 16, file);
+		bytes = fread(prebuffer, sizeof(char), len > 24 ? 24: len, file);
 		janus_pp_rtp_header *rtp = (janus_pp_rtp_header *)prebuffer;
 		JANUS_LOG(LOG_VERB, "  -- RTP packet (ssrc=%"SCNu32", pt=%"SCNu16", ext=%"SCNu16", seq=%"SCNu16", ts=%"SCNu32")\n",
 				ntohl(rtp->ssrc), rtp->type, rtp->extension, ntohs(rtp->seq_number), ntohl(rtp->timestamp));
 		if(rtp->csrccount) {
 			JANUS_LOG(LOG_VERB, "  -- -- Skipping CSRC list\n");
 			skip += rtp->csrccount*4;
 		}
+		audiolevel = -1;
+		rotation = -1;
 		if(rtp->extension) {
 			janus_pp_rtp_header_extension *ext = (janus_pp_rtp_header_extension *)(prebuffer+12);
```

Comment:
If this packet has CSRC field, then the extension is not at `prebuffer+12`.

Reply:
Actually there's no real need for that: this is just to print how many extensions there are, but we pass the start of the buffer to `janus_pp_rtp_header_extension_find`, which finds the actual extension we look for skipping CSRC as well. Anyway, it's true that we need to fix the check if we want to print the correct number there.

Reply:
Oh but we need that to update the skip value as well, so yeah, it was an important fix: thanks for spotting that!